auth/src/main/java/com/firebase/ui/auth/AuthUI.java

```diff
@@ -469,25 +546,265 @@ public Builder(@SupportedProvider @NonNull String providerId) {
+                } else {
+                    params = new Bundle();
+                }
+                setParams(params);
                 return this;
             }
 
+            @CallSuper
             @NonNull
-            public Builder setParams(@Nullable Bundle params) {
-                mParams = params;
+            public IdpConfig build() {
+                // Ensures deprecated Google provider builder backcompat
+                if (mProviderId.equals(GoogleAuthProvider.PROVIDER_ID)
+                        && getClass() == Builder.class
+                        && mParams.isEmpty()) {
+                    return new GoogleBuilder().build();
+                }
+
+                return new IdpConfig(mProviderId, mParams);
+            }
+        }
+
+        /**
+         * {@link IdpConfig} builder for the email provider.
+         */
+        public static final class EmailBuilder extends Builder {
+            public EmailBuilder() {
+                //noinspection deprecation taking a hit for the backcompat team
+                super(EmailAuthProvider.PROVIDER_ID);
+            }
+
+            /**
+             * Enables or disables creating new accounts in the email sign in flow.
+             * <p>
+             * Account creation is enabled by default.
+             */
+            @NonNull
+            public EmailBuilder setAllowNewAccounts(boolean allow) {
+                getParams().putBoolean(ExtraConstants.EXTRA_ALLOW_NEW_EMAILS, allow);
                 return this;
             }
+        }
 
+        /**
+         * {@link IdpConfig} builder for the phone provider.
+         */
+        public static final class PhoneBuilder extends Builder {
+            public PhoneBuilder() {
+                //noinspection deprecation taking a hit for the backcompat team
+                super(PhoneAuthProvider.PROVIDER_ID);
+            }
+
+            /**
+             * @param number the phone number in international format
+             * @see #setDefaultNumber(String, String)
+             */
             @NonNull
+            public PhoneBuilder setDefaultNumber(@NonNull String number) {
+                if (getParams().containsKey(ExtraConstants.EXTRA_COUNTRY_ISO)
+                        || getParams().containsKey(ExtraConstants.EXTRA_NATIONAL_NUMBER)) {
+                    throw new IllegalStateException("Cannot overwrite previously set phone number");
+                }
+
+                getParams().putString(ExtraConstants.EXTRA_PHONE, number);
+
+                return this;
+            }
+
+            /**
+             * Set the default phone number that will be used to populate the phone verification
+             * sign-in flow.
+             *
+             * @param iso    the phone number's country code
+             * @param number the phone number in local format
+             */
+            @NonNull
+            public PhoneBuilder setDefaultNumber(@NonNull String iso, @NonNull String number) {
```

Comment:
We need some validation/documentation on choosing an ISO code that we will accept.  I wouldn't know where to start getting one.

Reply:
Good point, we didn't have validation either before this PR. Let me see if I can cook something up. 😄

Reply:
Phone auth is so hard... I think I got it though. 😆